addons/event_crm/models/event_registration.py

```diff
@@ -0,0 +1,104 @@
+                        'partner_id': registration.partner_id.id,
+                        'email_from': registration.partner_id.email,
+                        'phone': registration.partner_id.phone,
+                        'mobile': registration.partner_id.mobile,
+                        'contact_name': registration.partner_id.name,
+                    })
+                else:
+                    lead = self.env['crm.lead'].search([('id', '=', registration.main_registration_id.lead_ids[0].id), ('contact_name', '=', registration.main_registration_id.partner_id.name)])
+                    if lead:
+                        lead.write({
+                            'partner_id': False,
+                            'name': "%s - %s" % (self.event_id.name, registration.main_registration_id.name),
+                            'contact_name': registration.main_registration_id.name,
+                            'email_from': registration.main_registration_id.email,
+                            'phone': registration.main_registration_id.phone,
+                            'mobile': registration.main_registration_id.mobile,
+                        })
+        else:
+            registrations = registration_ids if registration_ids else self
+            registrations = self.search([('id', 'in', registrations.ids)], order="id")
+            description = _("Other Participants:\n") if not registrations[0].main_registration_id else ""
+            for registration in registrations:
+                description += registration._get_lead_description()
+            registration_lead_values.update({
+                'description': description,
+                'registration_ids': registrations.ids,
+            })
+            if self.partner_id and self.partner_id != self.env.ref('base.public_partner'):
+                registration_lead_values.update({
+                    'name': "%s - %s" % (self.event_id.name, self.partner_id.name),
+                    'partner_id': self.partner_id.id,
+                    'email_from': self.partner_id.email,
+                    'phone': self.partner_id.phone,
+                    'mobile': self.partner_id.mobile,
+                    'contact_name': self.partner_id.name,
+                })
+            else:
+                registration_lead_values.update({
+                    'name': "%s - %s" % (self.event_id.name, self.name),
+                    'contact_name': self.name,
+                    'email_from': self.email,
+                    'phone': self.phone,
+                    'mobile': self.mobile,
+                })
+        return registration_lead_values
+
+    def _get_lead_description(self):
+        """
+        Build the description for the lead when a rule (per order) matchs for registrations.
+        """
+        return "\t%s %s %s\n" % (self.name, self.email if self.email else '', self.phone if self.phone else '')
```

Comment:
Why don't we set name as name, email as email_from and phone as phone ? and/or log a message ?